src/events/controller.py

```diff
@@ -153,6 +150,9 @@ def format_storages(self) -> None:
             initial_controllers=f"{self.charm.state.peer_cluster.bootstrap_unit_id}@{self.charm.state.peer_cluster.bootstrap_controller}:{self.charm.state.peer_cluster.bootstrap_replica_id}",
         )
 
+        # update status to add controller
+        self.charm.on.update_status.emit()
```

Comment:
Only difference that `update_status` does from `config_changed` is the CA bundle/truststore reloading. I would rather call config_changed here instead, to avoid the double `if self.charm.refresh_not_ready or not self.healthy` check. (one on status, and one right after)